### PR TITLE
[flutter_tools] update iOS deploy tests for best practices

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Uses less mocks and context

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/ios/devices_test.dart [hunk 1]
```diff
@@ -176,60 +176,6 @@ void main() {
       });
     }
 
-    group('ios-deploy wrappers', () {
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/ios/devices_test.dart]: Already tested in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/general.shard/ios/ios_device_install_test.dart#L79

**Review by christopherfujino — APPROVED** [review]
lgtm

**Comment by jmagman** [comment]
@jonahwilliams Can I merge this when the tree goes green?  I was about to add a test in the same place.

**Comment by jonahwilliams** [comment]
Better yet, I'll add the label :)

**Pull request merged**